### PR TITLE
CUMULUS-271 rules PUT request returns updated rule

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-271: Empty response body from rules PUT endpoint](https://bugs.earthdata.nasa.gov/browse/CUMULUS-271)

## Changes

* Rules PUT request returns updated rule

## Test Plan
Things that should succeed before merging.

- [x] Unit tests
- [x] Adhoc testing: tested in AWS
- [x] Update CHANGELOG
- [x] Run `./bin/eslint-ratchet` and verify that eslint errors have not increased
  - [ ] Commit `.eslint-ratchet-high-water-mark` if the score has improved

## Release PR

If this is a **release** PR, make sure you branch name starts with `release-` prefix, e.g. `release-v-1.1.2`.

Reviewers: @tag-your-friends

### CODE REVIEW

File: packages/api/endpoints/rules.js [hunk 1]
```diff
@@ -94,30 +95,47 @@ async function put(event) {
...
-    }
+    if (e instanceof RecordDoesNotExist) return cb({ message: 'Record does not exist' });
+    return cb(e);
   }
 
   // if rule type is onetime no change is allowed unless it is a rerun
   if (action === 'rerun') {
     await models.Rule.invoke(originalData);
-    return;
+    return cb(originalData);
```
Comment (abarciauskas-bgse) [inline packages/api/endpoints/rules.js]: I think this should be `return cb(null, originalData);`

File: packages/api/tests/test-endpoints-rules.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ test('POST creates a rule', (t) => {
 
 test('PUT updates a rule', (t) => {
   const updateEvent = {
-    body: JSON.stringify({state: 'ENABLED'}),
+    body: JSON.stringify({ state: 'ENABLED' }),
     pathParameters: {
       name: testRule.name
     },
```
[the remark this answers is not in the document]
Comment (abarciauskas-bgse) [reply packages/api/tests/test-endpoints-rules.js]: +1

[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/api/tests/test-endpoints-rules.js]: updated test and rules.js, and add a couple new tests.

File: packages/api/endpoints/rules.js [hunk 3]
```diff
@@ -94,30 +95,47 @@ async function put(event) {
...
   }
 
-  return model.update(originalData, data);
+  return model.update(originalData, data)
+    .then((d) => cb(null, d))
+    .catch((err) => cb(err));
```
[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/api/endpoints/rules.js]: eslint complains without (), maybe due to the override.

File: packages/api/endpoints/rules.js (hunk 1 above)
Comment (jennyhliu) [reply packages/api/endpoints/rules.js]: Yes.  Updated.

**Comment by jennyhliu** [comment]
@abarciauskas-bgse @laurenfrederick updated tests based on comments

**Pull request merged**